mlrun/feature_store/ingestion.py

```diff
@@ -279,7 +279,9 @@ def run_ingestion_job(name, featureset, run_config, schedule=None, spark_service
     featureset.save()
 
     # when running in server side we want to set the function db connection to the actual DB and not to use the httpdb
-    function.set_db_connection(featureset._get_run_db())
+    function.set_db_connection(
+        featureset._get_run_db(), is_api=mlrun.config.is_running_as_api()
```

Comment:
Why not simply change such that `set_db_connection` won't get the `is_api` parameter, and just do `self._is_api_server = mlrun.config.is_running_as_api()`?
After all, having to pass this parameter is error-prone (as we can see here), and it's no longer needed, since we have a reliable way to know that.